src/main/kotlin/gg/essential/elementa/markdown/MarkdownComponent.kt

```diff
@@ -46,6 +55,9 @@ class MarkdownComponent(
     }
 
     val drawables = DrawableList(this, emptyList())
+    private val offsets = HashMap<String, Float>()
+    val sectionOffsets: Map<String, Float>
+        get() = offsets
```

Comment:
Unless you rely on the specific implementation class, it is generally preferred to use `mutableMapOf<String, Float>()`.
More importantly though, we shouldn't be using a mutable map at all if we don't need the map to be internally mutable (which we don't). We can instead just have a single field with a private setter:
```suggestion
    var sectionOffsets: Map<String, Float> = emptyMap()
        private set
```
and atomically update it to the new offsets after the drawables have been layed out:
```
        sectionOffsets = drawables.filterIsInstance<HeaderDrawable>().associate { it.id to it.y }
```
This is not just easier to reason about because there's less mutability but it also has the advantage that `associate` is guaranteed to use a map which preserves the original iteration order, so one can iterate over the section offsets and they're already in the expected order.

Reply:
Assuming the markdown text can be very long with many different sections, would it not be better to store the header offsets while iterating over to lay out the y values rather than iterating separately?

Reply:
No, if you look at what the layout code itself actually does (and remember, this is layout, it's not even once per draw, just per layout), it should be apparent that a single extra iteration doesn't matter in the great scheme of things.